backend/src/apiserver/storage/run_store.go

```diff
@@ -262,7 +262,11 @@ func parseResourceReferences(resourceRefString sql.NullString) ([]*model.Resourc
 func (s *RunStore) CreateRun(r *model.RunDetail) (*model.RunDetail, error) {
 	if r.StorageState == "" {
 		r.StorageState = api.Run_STORAGESTATE_AVAILABLE.String()
+	} else if r.StorageState != api.Run_STORAGESTATE_AVAILABLE.String() &&
+		r.StorageState != api.Run_STORAGESTATE_ARCHIVED.String() {
+		return nil, util.NewInvalidInputError("Invalid value for StorageState field: %s.", r.StorageState)
```

Comment:
nit: %q will quote the invalid storage set.

Reply:
Good tip, done.